src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/Mock/MockConnection.cs

```diff
@@ -320,6 +325,12 @@ internal override ValueTask CloseAsync(long errorCode, CancellationToken cancell
                     state._serverErrorCode = errorCode;
                     DrainAcceptQueue(errorCode, -1);
                 }
+
+                foreach (KeyValuePair<long, MockStream.StreamState> kvp in state._streams)
```

Comment:
Nice, thanks for adding this. There are some QUIC tests that are disabled for the Mock provider specifically because we didn't previously support them. We should look to re-enable those -- obv not urgent though.